lib/Runtime/Library/BuiltIn/IndexOf.js

```diff
@@ -0,0 +1,48 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+"use strict";
+
+(function indexOf(searchElement, fromIndex) {
+
+    var k;
+
+    if (this == null) {
+        throw new TypeError('"this" is null or not defined');
+    }
+
+    var o = this;
+
+    var len = o.length >>> 0;
+
+    if (o.length <= 0) {
+        return -1;
+    }
+
+    var n = fromIndex | 0;
+
+    if (n >= len) {
+        return -1;
+    }
+
+    if (n >= 0) {
```

Comment:
well, maybe this is clearer separately like this, because before you were bounds checking, and here you're checking whether n is positive or negative

---
In reply to: [111842688](https://github.com/Microsoft/ChakraCore/pull/2845#discussion_r111842688) [](ancestors = 111842688)